src/redprl/categorical_judgment.sml

```diff
@@ -1,42 +1,70 @@
 structure RedPrlCategoricalJudgment :
 sig
   datatype ('sym, 'a) redprl_jdg =
-     EQ of ('a * 'a) * 'a
-   | TRUE of 'a
-   | EQ_TYPE of 'a * 'a
-   | SYNTH of 'a
+    (* `EQ_TYPE ((a, b), k)`:
+     *   `a` and `b` are equal types, taking into account the additional structures specified by `k`.
+     *   (That is, `EQ_TYPE ((a, b), KAN)` means they are equally Kan, not just having the same PER.)
+     *   The realizer is `TV` of sort `TRIV`.
+     *)
+     EQ_TYPE of ('a * 'a) * RedPrlKind.t
+   (* `EQ ((m, n), (a, k))`:
+    *   `EQ_TYPE ((a, a), k)` and `m` and `n` are related by the PER associated with `a`.
+     *   The realizer is `TV` of sort `TRIV`.
+    *)
+   | EQ of ('a * 'a) * ('a * RedPrlKind.t)
+    (* `TRUE (a, k)`:
+     *   `EQ_TYPE ((a, a), k)` and there exists a term `m` such that
+     *   `EQ ((m, m), (a, k))` is provable.
+     *   The realizer is such an `m` of sort `EXP`.
+     *)
+   | TRUE of ('a * RedPrlKind.t)
+    (* `SYNTH (m, k)`:
+     *   There exists a term `a` such that `EQ ((m, m), (a, k))` is provable.
+     *   Usually there is a canonical choice of `a`, but the canonicity is not required.
+     *   The realizer is such an `a` of sort `EXP`.
+     *)
+   | SYNTH of ('a * RedPrlKind.t)
+    (* `TERM tau`:
+     *   There exists some `m` of sort `tau`.
+     *   The realizer is such an `m` of sort `tau`.
+     *)
    | TERM of RedPrlSort.t
+    (* `DIM_SUBST ((p, r), m)`:
+     *   `p` should eventually be `DIM_REF p'` for some dimension term `p'`,
+     *   `r` is a dimension variable and `m` is an expression of sort `EXP`.
+     *   The realizer is the result of the substitution of `r` by `p'` in `m`.
+     *)
    | DIM_SUBST of ('a * 'sym) * 'a
```

Comment:
@jonsterling I wish that you could do an early review on the meaning of all the judgments.